engine/src/org/pentaho/di/www/TransformationMap.java

```diff
@@ -391,16 +375,8 @@ public void setSlaveServerConfig( SlaveServerConfig slaveServerConfig ) {
   /**
    * @return the hostServerSocketPortsMap
    */
-  public Map<String, List<SocketPortAllocation>> getHostServerSocketPortsMap() {
-    return hostServerSocketPortsMap;
-  }
-
-  /**
-   * @param hostServerSocketPortsMap
-   *          the hostServerSocketPortsMap to set
-   */
-  public void setHostServerSocketPortsMap( Map<String, List<SocketPortAllocation>> hostServerSocketPortsMap ) {
-    this.hostServerSocketPortsMap = hostServerSocketPortsMap;
+  public List<SocketPortAllocation> getHostServerSocketPorts( String hostname ) {
+    return Collections.unmodifiableList( hostServerSocketPortsMap.get( hostname ) );
```

Comment:
That is only necessary if you're not working with a List. My suggestion is to use a ConcurrentLinkedDeque, but that method returns a List. So my suggestion above is how you convert the Deque to a List and you do it by constructing a new ArrayList as I showed.